### PR TITLE
Upgrade to platformio 6.0

### LINKED ISSUE #1222: Migrate test suite to googletest v1.11
Our test suite no longer compiles with `googletest` v1.11.
A quick fix was implemented in #1221 to force `platformio` to download older version of `googletest` (v1.10).

A more sustainable solution is proposed by platformio support in https://community.platformio.org/t/googletest-v1-11-build-fail-on-native/27662 but requires quite a lot of work on the tests themselves.

**Starting points**
Check examples provided on platformio forums, and rework our unit test suite from there.

**Don't forget**
- [x] Put the issue in a milestone
- [x] Assign it to one of the projects
- [x] Add appropriate labels
- [x] Mention dependent/blocked issues if any

### LINKED ISSUE #1226: Controller and common tests fail in CircleCI
**Current behaviour/situation**
Tests fail in CI with:
```Platform Manager: Installing native
Downloading...
Unpacking...
Platform Manager: native @ 1.2.1 has been installed!
Platform Manager: Removing native @ 1.2.1
Platform Manager: native @ 1.2.1 has been removed!
Error: Development platform 'native' is not compatible with PlatformIO Core v5.2.5 and depends on PlatformIO Core ^6.
```

**Expected state/behaviour**
Test run succesfully on CI

**Don't forget**
- [x] Put the issue in a milestone
- [x] Assign it to one of the projects
- [x] Add appropriate labels
- [x] Mention dependent/blocked issues if any

### PR DESCRIPTION
**Pull request by asmodai27**
Platformio was upgraded to version 6.0, with only major change being the way unit tests are handled:
https://docs.platformio.org/en/latest/core/history.html

Followed migration guide and upgraded our installation scripts accordingly.
https://docs.platformio.org/en/latest/core/migration.html

Closes #1222
Closes #1226

This however leads to a new issue regarding coverage reports, which are now empty for common and controller code, but this is still better than failing CI.

I also fixed some cppcheck and clang-tidy warnings which became visible with this version of platformio.

# General checklist:

- [ ] I have performed a self-review, including - looked through the `Files changed` tab, browsed repository in my branch
- [x] I have made corresponding changes to the documentation - to reflect changes in code, electrical or mechanical design
- [x] All new documentation or graphics follow the [documentation style guide](https://github.com/RespiraWorks/Ventilator/wiki/Documentation-style-guide)
- [x] All new content is linked, easily discoverable, does not require too many clicks
- [x] I have reviewed any other relevant parts of our [contributor wiki](https://github.com/RespiraWorks/Ventilator/wiki) to ensure that this PR conforms to the standards
- [x] I have given the PR a descriptive name (prefixed with **WIP** if it is not yet ready for review)
- [x] I have tagged relevant reviewers

## Code checklist:

- [x] All new code follows our [code style](https://github.com/RespiraWorks/Ventilator/wiki/Code-style)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have described tests I used to verify my changes, instructions for reproducing them and any relevant configuration details.
- [x] I ran the unit test suite and verified that all tests pass - new and old, locally and on CI
- [x] I performed a clean build and verified that there were no warnings
- [x] I ran our static analysis tools and verified there were no warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] All source files have license headers

### CODE REVIEW

File: software/common/common.sh [hunk 1]
```diff
@@ -69,7 +69,7 @@ The following options are available:
                 [--cov]       - generate coverage reports
   cov_upload  Upload coverage reports to Codecov server
   unit        Builds and runs unit tests only (and generates coverage reports)
-                  <name>  - run specific unit test, may include wildcards, i.e. 'debug*'
+                  <name>  - run specific unit test, may include wildcards, i.e. '*common*'
```
Comment (martukas) [inline software/common/common.sh:72]: Why this change?

File: software/controller/controller.sh [hunk 2]
```diff
@@ -77,7 +77,7 @@ The following options are available:
                 [--no-checks] - do not run static checks (for CI)
                 [--cov]       - generate coverage reports
   unit        Builds and runs unit tests only (and generates coverage reports)
-                <name>  - run specific unit test, may include wildcards, i.e. 'debug*'
+                <name>  - run specific unit test, may include wildcards, i.e. '*debug*'
```
Comment (martukas) [inline software/controller/controller.sh:80]: but different here?

File: software/controller/platformio.ini [hunk 3]
```diff
@@ -58,6 +58,7 @@ build_flags =
 build_unflags =
   -std=gnu++11
   -std=gnu++14
+  -std=c++11
```
Comment (martukas) [inline software/controller/platformio.ini:61]: This is now needed in addition to gnu++11? But not c++14?
In fact, I thought we were even using 17 or even 20 features?

**Review by martukas — APPROVED** [review]
Just minor questions, none of them significant enough to require additional review.

File: software/common/common.sh (hunk 1 above)
Comment (asmodai27) [reply software/common/common.sh:72]: Filters now need to have this form to work, as unit test names usually starts with `test_`. I used `*common*` instead of `*debug*` because I wanted to give an example that actually works when using this script. Since then, the tests were restructured, and this example no longer works, will replace it with one that actually works.

File: software/controller/platformio.ini (hunk 3 above)
Comment (asmodai27) [reply software/controller/platformio.ini:61]: These are *un*-flags, meaning we specifically remove those from flags that are automatically added by platformio. I only added the one that platformio adds, and we need to remove (it caused build failures).
